streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/FlinkShimsProxy.scala

```diff
@@ -81,59 +83,106 @@ object FlinkShimsProxy extends Logger {
     })
   }
 
-  private[this] def getFlinkShimsClassLoader(flinkVersion: FlinkVersion): ClassLoader = {
+  // flink 1.12 1.13~1.14 1.15 parseSql class exist in different dependencies,
+  //need to load all flink-table dependencies compatible with different versions
+  def getVerifySqlLibClassLoader(flinkVersion: FlinkVersion): ClassLoader = {
+    logInfo(s"add  verify sql lib,flink version:  $flinkVersion")
+    VERIFY_SQL_CLASS_LOADER_CACHE.getOrElseUpdate(s"${flinkVersion.fullVersion}", {
+      val getFlinkTable: File => Boolean = _.getName.startsWith("flink-table")
+      // 1) flink/lib/flink-table*
+      val libTableURL = getFlinkHomeLib(flinkVersion.flinkHome, "lib", getFlinkTable)
+
+      // 2) After version 1.15 need add flink/opt/flink-table*
+      val optTableURL = getFlinkHomeLib(flinkVersion.flinkHome, "opt", getFlinkTable)
+      val shimsUrls = ListBuffer[URL](libTableURL ++ optTableURL: _*)
+
+      // 3) add only streampark shims jar
+      addShimsUrls(flinkVersion, shimsUrls, _.getName.startsWith("streampark-flink-shims"))
+      new ChildFirstClassLoader(
+        shimsUrls.toArray,
+        Thread.currentThread().getContextClassLoader,
+        getFlinkShimsResourcePattern(flinkVersion.majorVersion)
+      )
+    })
+  }
+
+  def addShimsUrls(flinkVersion: FlinkVersion, shimsUrls: ListBuffer[URL], filterJar: File => Boolean): Unit = {
```

Comment:
```suggestion
  def addShimsUrls(flinkVersion: FlinkVersion, addShimUrl: File => Unit): Unit = {
```

How about define the addShimUrl directly, and delete old addShimUrl?

getVerifySqlLibClassLoader caller:
```
      addShimsUrls(flinkVersion, file => {
        if (file.getName.startsWith("streampark-flink-shims")) {
          shimsUrls += file.toURI.toURL
        }
      })
```

getFlinkShimsClassLoader caller:
```
      addShimsUrls(flinkVersion, file => {
        shimsUrls += file.toURI.toURL
      })
```

Reply:
hi @1996fanrui  Thank you for your feedback.  Verifying sql dependencies also requires several classes in streampark's StreamPark-Flink-Shims module, so you need to add them as well

Reply:
As I understand, this comment just refactor, not change the logic, right?

Reply:
no, it's add a new logic. added to generate only the sql classloader, does not change the classloader for submitting jobs. Modifications were made to the common code during development that looked like refactoring.

Reply:
hi @1996fanrui i refactor it !